File: C++/Medium/WrongAnswer/range-product-queries-of-powers_6.cpp
class Solution {
public:
    vector<int> productQueries(int n, vector<vector<int>>& queries) {
        
        vector<int> powers;
        int p = 1;
        while (p <= n) {
            powers.push_back(p);
            p *= 2;
        }
        vector<int> ans;
        for (auto& q: queries) {
            int product = 1;
            for (int i = q[0]; i <= q[1]; ++i) {
                product = (product * powers[i]) % (1000000000 + 7);
            }
            ans.push_back(product);
        }
        return ans;
    }
};

File: Java/Medium/WrongAnswer/range-product-queries-of-powers_6.java
class Solution {
    public int[] productQueries(int n, int[][] queries) {
        
        int[] powers = new int[31];
        int[] products = new int[31];
        int[] answers = new int[queries.length];
        int j = 0;
        for (int i = 0; i < 31; i++) {
            powers[i] = (int) Math.pow(2, i);
        }
        for (int i = 0; i < queries.length; i++) {
            int left = queries[i][0];
            int right = queries[i][1];
            int product = 1;
            for (j = left; j <= right; j++) {
                if (products[j] != 0) {
                    product *= products[j];
                } else {
                    product *= powers[j];
                }
            }
            answers[i] = product;
            products[left] = product;
        }
        return answers;
    }
}